src/key_io.cpp

```diff
@@ -309,3 +311,33 @@ bool IsValidDestinationString(const std::string& str)
 {
     return IsValidDestinationString(str, Params());
 }
+
+std::pair<CPubKey, CPubKey> DecodeSilentData(const std::vector<unsigned char>& data)
+{
+
+    std::vector<unsigned char> scan_pubkey_data(data.begin(), data.begin() + 33);
+    CPubKey scan_pubkey{scan_pubkey_data};
+
+    std::vector<unsigned char> spend_pubkey_data(data.begin() + 33, data.end());
```

Comment:
37b0ca1c3d753cc61e0debf8ea59160bdf4127ec: alternatively you could scan for the exact pubkey length. That would tolerate future versions to add extra stuff to the address that's safe to ignore for older versions.

Reply:
Done